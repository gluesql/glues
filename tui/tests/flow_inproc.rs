mod common;
use common::AppTestExt as _;

use color_eyre::Result;
use insta::assert_debug_snapshot;

#[tokio::test]
async fn home_to_instant_quit_inproc() -> Result<()> {
    let (mut app, mut term) = common::setup_app_and_term().await?;

    // initial home screen
    app.draw_frame(&mut term)?;
    assert_debug_snapshot!("home_inproc", common::buffer_to_lines(&term));

    // open Instant (in-memory) notebook
    app.open_instant(&mut term).await?;
    assert_debug_snapshot!("instant_inproc", common::buffer_to_lines(&term));

    // quit with Ctrl+C
    let quit = app.ctrl('c').await;
    assert!(quit);

    Ok(())
}
