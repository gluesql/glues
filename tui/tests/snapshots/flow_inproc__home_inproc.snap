---
source: tui/tests/flow_inproc.rs
assertion_line: 29
expression: buffer_to_lines(&term)
snapshot_kind: text
---
[
    " Glues - TUI note-taking app offering complete data control and flexible storage options              \u{e0be} [?] Show keymap ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                           ████   ███                                                                   ",
    "                                          ██  ██   ██                                                                   ",
    "                                         ██        ██    ██  ██   ████    █████                                         ",
    "                                         ██        ██    ██  ██  ██  ██  ██                                             ",
    "                                         ██  ███   ██    ██  ██  ██████   ████                                          ",
    "                                          ██  ██   ██    ██  ██  ██          ██                                         ",
    "                                           █████  ████    ███ ██  ████   █████                                          ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                         ┌─────────────Open Notes─────────────┐                                         ",
    "                                         │                                    │                                         ",
    "                                         │   [1] Instant                      │                                         ",
    "                                         │   [2] Local                        │                                         ",
    "                                         │   [3] Git                          │                                         ",
    "                                         │   [4] MongoDB                      │                                         ",
    "                                         │   [5] CSV                          │                                         ",
    "                                         │   [6] JSON                         │                                         ",
    "                                         │   [h] Help                         │                                         ",
    "                                         │   [q] Quit                         │                                         ",
    "                                         │                                    │                                         ",
    "                                         └────────────────────────────────────┘                                         ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
    "                                                                                                                        ",
]
